test/cross-chain/cross-chain.js

```diff
@@ -88,13 +242,98 @@ contract("Cross-chain", () => {
+      const bridgeData = await homeColonyNetwork.getBridgeData(homeBridge.address);
+
+      expect(bridgeData.gas.toNumber()).to.equal(1000000);
+      expect(ethers.BigNumber.from(bridgeData.chainId).toHexString()).to.equal(ethers.BigNumber.from(foreignChainId).toHexString());
+      expect(bridgeData.setReputationRootHashBefore.toLowerCase()).to.equal(
+        `0xdc8601b3000000000000000000000000${foreignColonyNetwork.address.slice(
+          2
+          // eslint-disable-next-line max-len
+        )}000000000000000000000000000000000000000000000000000000000000006000000000000000000000000000000000000000000000000000000000000f42400000000000000000000000000000000000000000000000000000000000000044`.toLowerCase()
+      );
+      expect(bridgeData.setReputationRootHashAfter).to.equal(`0x00000000000000000000000000000000000000000000000000000000`);
+    });
+
+    it("mining bridge address is queryable", async () => {
+      let bridgeAddress = await homeColonyNetwork.getMiningBridgeAddress();
+      expect(bridgeAddress).to.equal(ADDRESS_ZERO);
+
+      bridgeAddress = await foreignColonyNetwork.getMiningBridgeAddress();
+      expect(bridgeAddress).to.equal(foreignBridge.address);
+    });
+
+    it("setBridgeData can only be called by the metacolony", async () => {
+      const tx = await foreignColonyNetwork.setBridgeData(ADDRESS_ZERO, 1, 0, "0x00", "0x00", "0x00", "0x00", "0x00", "0x00", { gasLimit: 1000000 });
+      await checkErrorRevertEthers(tx.wait(), "colony-caller-must-be-meta-colony");
+    });
+
+    it("setBridgeData can only set the mining chain bridge on a not-mining chain", async () => {
+      const tx = await foreignMetacolony.setBridgeData(ADDRESS_ZERO, 1, 0, "0x00", "0x00", "0x00", "0x00", "0x00", "0x00", { gasLimit: 1000000 });
+      await checkErrorRevertEthers(tx.wait(), "colony-network-can-only-set-mining-chain-bridge");
+    });
+
+    it("updating the bridge for a chain does not reset the bridged skill count", async () => {
+      const countBefore = await homeColonyNetwork.getBridgedSkillCounts(foreignChainId);
+      const tx = await homeMetacolony.setBridgeData(
+        homeBridge.address, // bridge address
+        foreignChainId, // chainid
+        1000000, // gas
+        "0x", // log before
+        "0x", // log after
+        `0x`, // skill before
+        "0x", // skill after
+        `0xdc8601b3000000000000000000000000${foreignColonyNetwork.address.slice(
+          2
+          // eslint-disable-next-line max-len
+        )}000000000000000000000000000000000000000000000000000000000000006000000000000000000000000000000000000000000000000000000000000f42400000000000000000000000000000000000000000000000000000000000000044`,
+        "0x00000000000000000000000000000000000000000000000000000000" // root hash after
+      );
+      await tx.wait();
+
+      const countAfter = await homeColonyNetwork.getBridgedSkillCounts(foreignChainId);
+      expect(countAfter).to.not.equal(0);
```

Comment:
Could do `be.zero` here

Reply:
Ethers BNs don't play nicely with that construction.